### PR TITLE
EOS-21367 Changes as per RE PR #472

### PR DESCRIPTION
**Pull request by vaibhavparatwar**
EOS-21367 Changes as per RE PR #472
Signed-off-by: Vaibhav Paratwar <vaibhav.paratwar@seagate.com>

### CODE REVIEW

**Review by knekrasov — APPROVED** [review]

**Review by johnbent — APPROVED** [review]

**Comment by mssawant** [comment]
@knekrasov , @shailesh-vaidya, @vaibhavparatwar,
Just one concern, here we are assuming that the puppet-agent package will include the relevant version of facter, thus we are restricting the given puppet-agent version, what if puppet stops distributing facter as part of it? can we handle such a situation?

**Comment by vaibhavparatwar** [comment]
@shailesh-vaidya  @knekrasov - what is your view on @mssawant's concern? Community is blocked on this - pls check this GitHub issue - https://github.com/Seagate/cortx-hare/issues/1665

**Comment by daniarherikurniawan** [comment]
@vaibhavparatwar, @mssawant, @knekrasov   With regards to issue https://github.com/Seagate/cortx-hare/issues/1665, I tried out this PR and I get another error 
```sh
error: Failed dependencies:
	puppet-agent >= 6.13.0 is needed by cortx-hare-2.0.0-1_git47b95c8.el7.x86_64
``` 
which is expected, right?

I tried to run `make` and `make install` on hare directory, but the puppet is still not installed automatically.

So, which script will install this `puppet-agent` at which step (w.r.t the hare [README](https://github.com/Seagate/cortx-hare/blob/main/README.md))?
Perhaps the README should be edited as well, to include `puppet` installation. My main task is to make sure that the README is bulletproof :) 
Thanks!

**Comment by mssawant** [comment]
@vaibhavparatwar, kindly address @daniarherikurniawan's comment.

**Comment by vaibhavparatwar** [comment]
> @vaibhavparatwar, @mssawant, @knekrasov With regards to issue #1665, I tried out this PR and I get another error
> 
> ```shell
> error: Failed dependencies:
> 	puppet-agent >= 6.13.0 is needed by cortx-hare-2.0.0-1_git47b95c8.el7.x86_64
> ```
> 
> I tried to run `make` and `make install` on hare directory, but the puppet is still not installed automatically.
> 
> So, which script will install this `puppet-agent` at which step (w.r.t the hare [README](https://github.com/Seagate/cortx-hare/blob/main/README.md))?
> Perhaps the README should be edited as well, to include `puppet` installation. My main task is to make sure that the README is bulletproof :)
> Thanks!

Hi @daniarherikurniawan The script is part of Cortx-RE - check this. - https://github.com/Seagate/cortx-re/pull/472/files#diff-e2aa453f6758b0dcb4f493b2e1c6ba48a30ada3cc2023093ad34f24a9a7a4fff  actually it runs as part of full stack deployment. 
We may need to add this in hare README. Will discuss with @knekrasov and @mssawant for this.

**Comment by daniarherikurniawan** [comment]
Thanks @vaibhavparatwar ! 
I believe the cortx-re is a private repo that we (community) don't have access to. Looking forward to seeing the update at the public cortx repo so that we can safely satisfy all dependencies!

**Comment by vaibhavparatwar** [comment]
@daniarherikurniawan let me get some idea on this from @knekrasov , @mssawant and @shailesh-vaidya - Like this dependency are there any other dependencies as well, which are not part of README? so we address them all together in hare README?

**Comment by johnbent** [comment]
@knekrasov and @mssawant and @vaibhavparatwar , what is the status of this one please?

File: hare.spec [hunk 1]
```diff
@@ -47,7 +47,7 @@ BuildRequires: python36-pip
 BuildRequires: python36-setuptools
 
 Requires: consul >= 1.7.0, consul < 1.10.0
-Requires: facter >= 3.14.8
+Requires: puppet-agent >= 6.13.0
```
Comment (andriytk) [inline hare.spec:50]: I'm afraid this won't fix the issue, since `puppet-agent` is Seagate's custom rpm which is not available for external users.

Comment (mssawant) [reply hare.spec:50]: @andriytk, I can find puppet-agent rpms at https://yum.puppetlabs.com/puppet/el/7/x86_64/ also at (https://puppet.com/docs/puppet/7/install_agents.html#install_agents)  , I think within Seagate it is installed from Seagate's private repo but if we can document the appropriate external repo to be configured for the community then I think it should be fine, no? cc @vaibhavparatwar, @knekrasov

Comment (andriytk) [reply hare.spec:50]: Nice, didn't know that. Do they provide the needed version of facter?

**Comment by vaibhavparatwar** [comment]
raised PR #1743 for this. Closing this here.

**Pull request closed without merging**